### PR TITLE
UpdateBoundingVolume MT during Unit::SlowUpdate

### PR DESCRIPTION
**Pull request by Beherith**
Try to save half a millisec per frame lategame by performing  this MT'd.

Ive temporarily added a config int SlowUpdateBoundingVolumeMT to toggle this ingame for comparison only. Otherwise toggling this ingame may lead to a desync.

### CODE REVIEW

File: rts/Sim/Units/UnitHandler.cpp [hunk 1]
```diff
@@ -351,16 +355,42 @@ void CUnitHandler::SlowUpdateUnits()
...
+			unit->SanityCheck();
+		}
+	}
+	else {
+		{
+			ZoneScopedN("Sim::Unit::SlowUpdateST");
+			for (size_t i = idxBeg; i < idxEnd; ++i) {
+				CUnit* unit = activeUnits[i];
+
+				unit->SanityCheck();
```
Comment (sprunk) [inline rts/Sim/Units/UnitHandler.cpp:378]: Does it need this sanity check if there's also one in the MT section?

Comment (Beherith) [reply rts/Sim/Units/UnitHandler.cpp:378]: IIRC that sanity check compiles to a no-op on release. I think its thus not a bad idea to have it both before and after slowupdate, similarly to ST implementation

**Comment by Beherith** [comment]
> I checked before: It can't cause a desync, No need to have that runtime switch (other than for bench-marking purposes).
I wasnt completely sure if that was the case given that the result of			
`
unit->SlowUpdate();
unit->SlowUpdateWeapons();	
`
could possibly depend on another unit's bounding volume, which now is updated later in batch.

**Review by sprunk — APPROVED** [review]

**Comment by lhog** [comment]
Pls move the `for_mt(idxBeg, idxEnd, [&](const int i)` block before the ST block.
As far as I can tell
```
		unit->SlowUpdate();
		unit->SlowUpdateWeapons();
```
Don't move pieces or execute any animation, so it should be a safe move.

**Comment by lhog** [comment]
Yep, you're right.

**Comment by Beherith** [comment]
It doesnt matter really which order it is, in fact, UpdateBoundingVolume before might actually better because units surely arent dead yet then. hmm @lhog ?

**Comment by lhog** [comment]
I think you better keep it as is. But add `if (unit->isDead)` early exit.

**Comment by Beherith** [comment]
Damn I didnt commit my change, and now broke my repo on a failed merged :(

**Comment by 6AKU66** [comment]
> Seems like SlowUpdateWeapons can execute scripts: image

Where i can see such stuff?

**Comment by lhog** [comment]
> > Seems like SlowUpdateWeapons can execute scripts: image
> 
> Where i can see such stuff?

MS Visual Studio.

**Pull request merged**